src/backend/glutin/mod.rs

```diff
@@ -0,0 +1,271 @@
+    /// Create a new glium `Display`.
+    ///
+    /// Performs a compatibility check to make sure that all core elements of glium are supported
+    /// by the implementation.
+    pub fn from_gl_window(gl_window: glutin::GlWindow) -> Result<Self, IncompatibleOpenGl> {
+        Self::with_debug(gl_window, Default::default())
+    }
+
+    /// Create a new glium `Display`.
+    ///
+    /// This function does the same as `build_glium`, except that the resulting context
+    /// will assume that the current OpenGL context will never change.
+    pub unsafe fn unchecked(gl_window: glutin::GlWindow) -> Result<Self, IncompatibleOpenGl> {
+        Self::unchecked_with_debug(gl_window, Default::default())
+    }
+
+    /// The same as the `new` constructor, but allows for specifying debug callback behaviour.
+    pub fn with_debug(gl_window: glutin::GlWindow, debug: debug::DebugCallbackBehavior)
+        -> Result<Self, IncompatibleOpenGl>
+    {
+        Self::new_inner(gl_window, debug, true)
+    }
+
+    /// The same as the `unchecked` constructor, but allows for specifying debug callback behaviour.
+    pub unsafe fn unchecked_with_debug(
+        gl_window: glutin::GlWindow,
+        debug: debug::DebugCallbackBehavior,
+    ) -> Result<Self, IncompatibleOpenGl>
+    {
+        Self::new_inner(gl_window, debug, false)
+    }
+
+    fn new_inner(
+        gl_window: glutin::GlWindow,
+        debug: debug::DebugCallbackBehavior,
+        checked: bool,
+    ) -> Result<Self, IncompatibleOpenGl>
+    {
+        let gl_window = Rc::new(RefCell::new(gl_window));
+        let glutin_backend = GlutinBackend(gl_window.clone());
+        let framebuffer_dimensions = glutin_backend.get_framebuffer_dimensions();
+        let context = try!(unsafe { context::Context::new(glutin_backend, checked, debug) });
+        Ok(Display {
+            gl_window: gl_window,
+            context: context,
+            last_framebuffer_dimensions: Cell::new(framebuffer_dimensions),
+        })
+    }
+
+    /// Rebuilds the Display's `GlWindow` with the given window and context builders.
+    ///
+    /// This method ensures that the new `GlWindow`'s `Context` will share the display lists of the
+    /// original `GlWindow`'s `Context`.
+    pub fn rebuild(
+        &self,
+        window_builder: glutin::WindowBuilder,
+        context_builder: glutin::ContextBuilder,
+        events_loop: &glutin::EventsLoop,
+    ) -> Result<(), DisplayCreationError>
+    {
+        // Share the display lists of the existing context.
+        let new_gl_window = {
+            let gl_window = self.gl_window.borrow();
+            let context_builder = context_builder.with_shared_lists(gl_window.context());
+            try!(glutin::GlWindow::new(window_builder, context_builder, events_loop))
+        };
+
+        // Replace the stored GlWindow with the new one.
+        let mut gl_window = self.gl_window.borrow_mut();
+        std::mem::replace(&mut (*gl_window), new_gl_window);
+
```

Comment:
You need to call `self.context.rebuild()` because [what it does is important](https://github.com/glium/glium/blob/24984cea9380c96cae0a323ca7d2706c7a288aad/src/context/mod.rs#L242-L273).